lib/order.rb

```diff
@@ -0,0 +1,64 @@
+require 'csv'
+require 'pry'
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    valid_statuses = %i[pending paid processing shipped complete]
+    raise ArgumentError, "Invalid fulfillment_status" unless valid_statuses.include?(@fulfillment_status)
+  end
+
+  def total
+    product_sum = @products.values.sum
+    sum_plus_tax = product_sum * 1.075
+    return sum_plus_tax.round(2)
+  end
+
+  def add_product(name, price)
+    if @products.keys.include?(name)
+      raise ArgumentError, "Duplicate product"
+    else
+      @products[name] = price
+    end
+  end
+
+  def self.product_parse(products)
+    product_hash = {}
+    products_split = products.split(';')
+    products_split.each do |product|
```

Comment:
Good work on this tricky parsing logic. I also really appreciate that you broke it out into a separate method, this makes the flow of the program much easier to understand.